### DISCUSSION TITLE
What is the difference of using reflection and Cecil for modifying assemblies?

### DISCUSSION THREAD

**Discussion by mehdihadeli** [jbevain/cecil#891] [Q&A]
Hi,
What is the difference of using reflection and Cecil for modifying assemblies?

**Comment by adrianoc** [comment]
I'd say that Cecil will give you much more flexibility since reflection does not allow you to modify existing assemblies.

That said, reflection does allow you to emit types/methods/properties/fields/events using `TypeBuilder` (and related types) both in the executing assembly (i.e, at runtime) or in a newly created one. Note that injecting types/members in the executing assembly will not work in platforms that relies on AOT (for instance, iOS);
